e2e/test/scenarios/dashboard-cards/click-behavior.cy.spec.js

```diff
@@ -0,0 +1,716 @@
+        },
+      );
+    });
+
+    it("allows updating multiple dashboard filters", () => {
+      const dashboardDetails = {
+        parameters: [DASHBOARD_FILTER_TEXT, DASHBOARD_FILTER_TIME],
+      };
+
+      cy.createQuestionAndDashboard({ questionDetails, dashboardDetails }).then(
+        ({ body: card }) => {
+          visitDashboard(card.dashboard_id);
+          editDashboard();
+
+          getDashboardCard().realHover().icon("click").click();
+          cy.get("aside").findByText("Update a dashboard filter").click();
+          addTextParameter();
+          addTimeParameter();
+          cy.get("aside").button("Done").click();
+
+          saveDashboard();
+
+          clickLineChartPoint();
+          cy.findAllByTestId("field-set").should("have.length", 2);
+          cy.findAllByTestId("field-set").should("contain.text", POINT_COUNT);
+          cy.findAllByTestId("field-set").should(
+            "contain.text",
+            POINT_CREATED_AT_FORMATTED,
+          );
+          cy.location("search").should(
+            "eq",
+            `?${DASHBOARD_FILTER_TEXT.slug}=${POINT_COUNT}&${DASHBOARD_FILTER_TIME.slug}=${POINT_CREATED_AT}`,
+          );
+        },
+      );
+    });
+  });
+});
+
+/**
+ * @param {string} value
+ * @returns string
+ *
+ * @see https://docs.cypress.io/api/commands/type#Arguments
+ */
+const escapeCypressCurlyBraces = value => value.replaceAll("{", "{{}");
+
+/**
+ * This function exists to work around custom dynamic anchor creation.
+ * @see https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/dom.js#L301-L312
+ *
+ * WARNING: For the assertions to work, ensure that a click event occurs on an anchor element afterwards.
+ */
+const onNextAnchorClick = callback => {
+  cy.window().then(window => {
+    const originalClick = window.HTMLAnchorElement.prototype.click;
+
+    window.HTMLAnchorElement.prototype.click = function () {
+      callback(this);
+      window.HTMLAnchorElement.prototype.click = originalClick;
+    };
+  });
+};
+
+/**
+ * Duplicated from metabase/lib/card because Cypress can't handle import from there.
+ *
+ * @param {string} value
+ * @returns object
+ */
+const deserializeCardFromUrl = serialized =>
+  JSON.parse(b64hash_to_utf8(serialized));
+
+const clickLineChartPoint = () => {
+  cy.findByTestId("dashcard")
+    .get("circle.dot")
+    .eq(POINT_INDEX)
+    /**
+     * calling .click() here will result in clicking both
+     *     g.voronoi > path[POINT_INDEX]
+     * and
+     *     circle.dot[POINT_INDEX]
+     * To make it worse, clicks count won't be deterministic.
+     * Sometimes we'll get an error that one element covers the other.
+     * This problem prevails when updating dashboard filter,
+     * where the 2 clicks will cancel each other out.
+     **/
+    .then(([circle]) => {
+      const { left, top } = circle.getBoundingClientRect();
+      cy.get("body").click(left, top);
+    });
+};
+
+const addDashboardDestination = () => {
```

Comment:
These helpers will also be used in [table tests](https://github.com/metabase/metabase/issues/35035)